client/util/dates.utils.js

```diff
@@ -5,7 +5,7 @@ import moment from 'moment';
 ensure that all adjacent date ranges are merged into one. (eg. 17-21 and 22-25 => 17-25)*/
 export const dateRangeReducer = (dates) => {
```

Comment:
im not shure about the logic... try refink... maybe you can use a haigher order function or a better array prototype... i dont know